src/UpgradeRule/PHP/Visitor/Warnings/WarningsVisitor.php

```diff
@@ -64,7 +65,10 @@ public function enterNode(Node $node)
             if ($this->matchesSpec($node, $spec)) {
                 // Only match first spec that matches this node
                 $this->rewriteWithSpec($node, $spec);
-                $this->addWarning($node, $spec);
+                if ($spec->getMessage() && !$node instanceof Property) {
```

Comment:
Does `!$node instanceof Property` evaluate correctly to "node is not an instance of Property"? Or does it evaluate to "falsy $node is an instance of Property"? Might be worth adding brackets to be sure

Reply:
`!$node instanceof Property` has implied brackets `!($node instanceof Property)`.

Reply:
Cool, thanks (TIL)

Reply:
@maxime-rainville yep, warnings would previously always show, but with these changes you should now be able to add warning props without requiring warning messages (to accomodate scenarios where we just want to update visibility).

The `!$node instanceof Property` check is just there to prevent a duplicate warning message as it will still trigger a warning with the `PropertyProperty` class.